src/mock.rs

```diff
@@ -22,16 +25,17 @@ use rwlock::RwLock;
 /// The tree store is internally represented with a `HashMap`.  This structure
 /// is exposed for use only by downstream crates' tests, and it should obviously
 /// not be used in production.
+#[derive(Debug, Clone)]
 pub struct MockTreeStore {
     #[allow(clippy::type_complexity)]
-    data: RwLock<(HashMap<NodeKey, Node>, BTreeSet<StaleNodeIndex>)>,
+    data: Arc<RwLock<(HashMap<NodeKey, Node>, BTreeSet<StaleNodeIndex>)>>,
```

Comment:
Adding the `Arc` here allows using the `MockTreeStore` by value in several places at once, which makes writing the tests easier (since it can be simultaneously in a `WriteOverlay` and also *used* by the `WriteOverlay` to commit to).